django-ory-cloud/playwright.config.ts

```diff
@@ -0,0 +1,109 @@
+import type { PlaywrightTestConfig } from '@playwright/test'
+import { devices } from '@playwright/test'
+
+/**
+ * Read environment variables from file.
+ * https://github.com/motdotla/dotenv
+ */
+// require('dotenv').config();
+
+/**
+ * See https://playwright.dev/docs/test-configuration.
+ */
+const config: PlaywrightTestConfig = {
+  testDir: './tests',
+  /* Maximum time one test can run for. */
+  timeout: 30_000,
+  expect: {
+    /**
+     * Maximum time expect() should wait for the condition to be met.
+     * For example in `await expect(locator).toHaveText();`
+     */
+    timeout: 1_000
+  },
+  /* Run tests in files in parallel */
+  fullyParallel: true,
+  /* Fail the build on CI if you accidentally left test.only in the source code. */
+  forbidOnly: !!process.env.CI,
+  /* Retry on CI only */
+  retries: process.env.CI ? 2 : 0,
+  /* Opt out of parallel tests on CI. */
+  workers: process.env.CI ? 1 : undefined,
```

Comment:
I'm still uncomfortable with using different settings for parallelism in different environments here. If parallelism works, then let's use it on CI as well since it makes CI run faster (which is meaningful). If parallelism doesn't work, we shouldn't run it locally either. I cannot see a scenario where the current setting would make sense. Am I missing something?

```suggestion
  /* the number of workers, undefined for parallel tests */
  workers: 1,
```

Reply:
Sure, this is just the default setting when creating a new playwright config. 
I figured its there for some reason, but did not investigate.

Reply:
I have read through https://playwright.dev/docs/test-parallel and don't see any technical reason in there for limiting parallelism. It doesn't change how tests run. Playwright always runs tests in worker processes. Parallelism just means having more than one worker process at a time. There are a few bug reports on Playwright's issue tracker that parallel tests are flaky, but they all come from problems with the setup of the respective Playwright users.

I assume this setting is the safest default setting to minimize customer support tickets to the vendor. When self-hosting, the CI server is often a single computer. This machine runs all tests from all teams. If one team uses Playwright and Playwright uses all the cores on that machine, then all the other tests from other teams that are also running at that time slow down to a crawl, resulting in false positive test failures from timeouts. People might yell at Microsoft for creating a product that breaks their CI server with the standard configuration. On the other hand, the Playwright folks also don't want to disable parallelism in the default config. So the default config enables parallelism locally and disables it on CI as a compromise. That's safe for everybody, and teams who know what they are doing can enable it on CI if they want to. 

We use scalable cloud infrastructure for CI. Each of our tests gets their own dedicated virtual machine. Our CI server isn't a bottleneck, and I think we can safely enable parallelism on CI.